compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1163,6 +1163,426 @@ TR::Register *OMR::X86::TreeEvaluator::cstoreEvaluator(TR::Node *node, TR::CodeG
+      generateInstruction(CMPSD, node, deps, cg);
+      generateLabelInstruction(JNE4, node, notEqualLabel, cg);
+
+      //QWord Compare
+      generateLabelInstruction(LABEL, node, qWordCompareLabel, cg);
+      generateInstruction(REPECMPSQ, node, deps, cg);
+      generateLabelInstruction(JNE4, node, notEqualLabel, cg);
+   }
+   else 
+   {
+      generateLabelInstruction(LABEL, node, dWordCompareLabel, cg);
+      generateInstruction(REPECMPSD, node, deps, cg);
+      generateLabelInstruction(JNE4, node, notEqualLabel, cg);
+   }
+
+   cg->stopUsingRegister(counterRegister);
+   cg->stopUsingRegister(src1AddrRegister);
+   cg->stopUsingRegister(src2AddrRegister);
+
+   //Result 
+   generateLabelInstruction(LABEL, node, equalLabel, cg);
+   generateRegImmInstruction(MOVRegImm4(), node, resultRegister, 0, cg); //Equal
+   generateLabelInstruction(JMP4, node, endResultLabel, cg);
+
+   generateLabelInstruction(LABEL, node, notEqualLabel, cg);
+   generateLabelInstruction(JL4, node, lessThanLabel, cg);
+   generateRegImmInstruction(MOVRegImm4(), node, resultRegister, 2, cg); //Greater Than
+   generateLabelInstruction(JMP4, node, endResultLabel, cg);
+
+   generateLabelInstruction(LABEL, node, lessThanLabel, cg);
+   generateRegImmInstruction(MOVRegImm4(), node, resultRegister, 1, cg); //Less Than`
+   
+   deps->stopAddingConditions();
+
+   generateLabelInstruction(LABEL, node, endResultLabel, deps, cg);
+   endResultLabel->setEndInternalControlFlow();
+
+   node->setRegister(resultRegister);
+
+   cg->decReferenceCount(src1AddrNode);
+   cg->decReferenceCount(src2AddrNode );
+   cg->decReferenceCount(lengthNode );
+
+   return resultRegister;
+   }
+
+static TR::Register *REPArraycmpLenEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Register *resultRegister = cg->allocateRegister(TR_GPR);
+   TR::Node *src1AddrNode = node->getChild(0);
+   TR::Node *src2AddrNode = node->getChild(1);
+   TR::Node *lengthNode = node->getChild(2);
+   
+   int32_t byteLen = 4;
+   if (TR::Compiler->target.is64Bit())
+      byteLen = 8;
+
+   TR::Register *tempSrc1AddrRegister = cg->evaluate(src1AddrNode);;
+   TR::Register *tempSrc2AddrRegister = cg->evaluate(src2AddrNode);
+   TR::Register *tempArrayLenRegister = cg->evaluate(lengthNode);
+
+   TR::Register *src1AddrRegister = cg->allocateRegister();
+   TR::Register *src2AddrRegister = cg->allocateRegister();
+   TR::Register *arrayLenRegister = cg->allocateRegister();
+   TR::Register *counterRegister = cg->allocateRegister();
+
+   TR::LabelSymbol *startLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *residualLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *notEqualLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *backtrackLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *equalLabel = generateLabelSymbol(cg);
+   TR::LabelSymbol *endResultLabel = generateLabelSymbol(cg);
+
+   //Move into registers to for use
```

Comment:
These moves are done regardless of the lifetime of the original register (i.e., if the ref count is one you don't need to copy it).  I think a `gprClobberEvaluate` instead of an `evaluate` on these nodes is what you want, as you'll get the copy done for you.